src/Concordium/Client/RWLock.hs

```diff
@@ -0,0 +1,201 @@
+module Concordium.Client.RWLock
+  ( RWLock(..)
```

Comment:
Do we need to export the fields? We should not need that, so unless I am missing something export just the type.